Sources/MapboxCoreNavigation/CoreNavigationNavigator.swift

```diff
@@ -142,6 +143,7 @@ class Navigator {
         navigator.setElectronicHorizonObserverFor(self)
         navigator.addObserver(for: self)
         navigator.setFallbackVersionsObserverFor(self)
+        historyRecorder?.startRecording()
```

Comment:
The relevant change in MapboxNavigationNative clearly expects the history recorder to be started explicitly at some point before dumping (or “stopping”) history, so I’ve added a bit here to start recording basically as soon as possible. In principle, we could start recording more lazily, perhaps based on some application-defined behavior, but we don’t have a convenient place to do that, so for now I’m just preserving the previous behavior.

/cc @mskurydin @DmitryAk